### ISSUE TITLE
Getting caught up on unavailable Hetzner server?

### ISSUE THREAD

**Issue by JulianGro** [testflows/TestFlows-GitHub-Hetzner-Runners#36]
Our CI got caught up on something again. This time I believe it tried to provision a CX52 server, which aren't currently available anymore. I had to restart it to get it to provision the servers which are available.

Everything gets logged to syslog, but I don't actually know how to get anything meaningful out of there. If I just scroll through the log using journalctl, it doesn't allow scrolling far enough back.

**Comment by vzakaznikov** [comment]
Logs messages are also written to a rotating log file https://github.com/testflows/TestFlows-GitHub-Hetzner-Runners/blob/main/testflows/github/hetzner/runners/logger.py#L171.
You should be able to find them in the temp directory.

**Comment by vzakaznikov** [comment]
The issue is most likely due to https://github.com/testflows/TestFlows-GitHub-Hetzner-Runners/blob/main/testflows/github/hetzner/runners/scale_up.py#L929 where we abort on the first exception.
Instead we should be waiting for all of the futures to complete.

**Comment by vzakaznikov** [comment]
> The issue is most likely due to https://github.com/testflows/TestFlows-GitHub-Hetzner-Runners/blob/main/testflows/github/hetzner/runners/scale_up.py#L929 where we abort on the first exception. Instead we should be waiting for all of the futures to complete.

No, we are waiting for all futures becuase ignore_fail=True so most likely not an issue. When servers are not available but a job requests it we will keep trying.
